src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+
+        List<? extends Machine> types = new ArrayList<>();
+        if (Bulldozer.class.equals(type)) {
+            types = new BulldozerProducer().get();
+        }
+        if (Excavator.class.equals(type)) {
+            types = new ExcavatorProducer().get();
+        }
+        if (Truck.class.equals(type)) {
+            types = new TruckProducer().get();
+        }
+        return (List<Machine>) types;
```

Comment:
You can write like:
```suggestion
        return new ArrayList(types);
```
to avoid casting